sdlf-utils/glue-jobs-deployer/glue-job.yaml

```diff
@@ -0,0 +1,97 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "Engineering <<pipeline-dataset>> Glue Job"
+
+Parameters:
+    pPipelineBucket:
+      Description: "The artifactory bucket used by CodeBuild and CodePipeline"
+      Type: "AWS::SSM::Parameter::Value<String>"
+      Default: "/SDLF/S3/ArtifactsBucket"
+    pJobFiles:
+      Type: String
+      Description: The .sql and .json files from the jobs.
+    pPipeline:
+      Type: String
+      Description: The pipeline name (i.e. analyticspipeline).
+    pTeam:
+      Type: String
+      Description: The team name (i.e. engineering)
+    pDataset:
+      Type: String
+      Description: The dataset name (i.e. sampledataset)
+    pAllocatedCapacity:
+      Type: Number
+      Description: The glue job allocated capacity
+
+Resources:
+  rGlueRole:
+    Type: "AWS::IAM::Role"
+    Properties: 
+      RoleName: !Sub 'sdlf-${pPipeline}-${pDataset}-glue-role'
+      Path: "/service-role/"
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - "glue.amazonaws.com"
+            Action: 
+              - "sts:AssumeRole"
+      ManagedPolicyArns: 
+        - "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+        - "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+        - "arn:aws:iam::aws:policy/CloudWatchLogsFullAccess"
+        - "arn:aws:iam::aws:policy/AWSLakeFormationDataAdmin"
+      Policies:
+        - PolicyName: !Sub 'sdlf-${pTeam}-${pPipeline}-${pDataset}-glue-job-policy'
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Action:
+                  - "s3:*"
+                  - "glue:*"
+                  - "dynamodb:*"
+                  - "redshift:*"
+                  - "logs:*"
+                  - "cloudwatch:*"
```

Comment:
Aren't the managed policies and the actions overlapping here?
For example, `AmazonS3FullAccess` and `"s3:*"` are equivalent so perhaps we don't need the second one? Same for Glue and CloudWatch. Also, would it be possible to limit Redshift and DynamoDB policies to least privilege? `"*"` is a bit too open

Reply:
Make sense, the role was fixed, thank you!